pkg/clusteragent/admission/mutate/agent_sidecar/util.go

```diff
@@ -8,111 +8,42 @@
-					SecretKeyRef: &corev1.SecretKeySelector{
-						Key: "api-key",
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: "datadog-secret",
-						},
-					},
-				},
-			},
-			{
-				Name:  "DD_SITE",
-				Value: ddSite,
-			},
-			{
-				Name:  "DD_CLUSTER_NAME",
-				Value: config.Datadog.GetString("cluster_name"),
-			},
-			{
-				Name: "DD_KUBERNETES_KUBELET_NODENAME",
-				ValueFrom: &corev1.EnvVarSource{
-					FieldRef: &corev1.ObjectFieldSelector{
-						APIVersion: "v1",
-						FieldPath:  "spec.nodeName",
-					},
-				},
-			},
-		},
-		Image:           "datadog/agent",
-		ImagePullPolicy: corev1.PullIfNotPresent,
-		Name:            agentSidecarContainerName,
-		Resources: corev1.ResourceRequirements{
-			Requests: map[corev1.ResourceName]resource.Quantity{
-				"memory": resource.MustParse("256Mi"),
-				"cpu":    resource.MustParse("200m"),
-			},
-			Limits: map[corev1.ResourceName]resource.Quantity{
-				"memory": resource.MustParse("256Mi"),
-				"cpu":    resource.MustParse("200m"),
-			},
-		},
-	}
-
-	return agentContainer
-}
-
-////////////////////////////////
-//                            //
-//     Provider Overrides     //
-//                            //
-////////////////////////////////
-
-// ProviderIsSupported indicates whether the provider is supported by agent sidecar injection
-func ProviderIsSupported(provider string) bool {
-	switch provider {
-	case providerFargate:
-		return true
-	default:
-		return false
+	if container == nil {
+		return fmt.Errorf("can't apply environment overrides to nil container")
 	}
-}
 
-func applyProviderOverrides(container *corev1.Container) {
-	provider := config.Datadog.GetString("admission_controller.agent_sidecar.provider")
-
-	if !ProviderIsSupported(provider) {
-		log.Errorf("unsupported provider: %v", provider)
-		return
+	for _, envVarOverride := range extraEnv {
+		// Check if the environment variable already exists in the container
+		var found bool
+		for i, envVar := range container.Env {
+			if envVar.Name == envVarOverride.Name {
+				// Override the existing environment variable value
+				container.Env[i] = envVarOverride
+				found = true
+				break
+			}
+		}
+		// If the environment variable doesn't exist, add it to the container
+		if !found {
+			container.Env = append(container.Env, envVarOverride)
+		}
 	}
 
-	switch provider {
-	case providerFargate:
-		applyFargateOverrides(container)
-	}
+	return nil
 }
 
-func applyFargateOverrides(container *corev1.Container) {
+// withResourceLimits applies the resource limits overrides to the container
+func withResourceLimits(container *corev1.Container, resourceLimits corev1.ResourceRequirements) error {
 	if container == nil {
-		return
+		return fmt.Errorf("can't apply resource requirements overrides to nil container")
 	}
-
-	container.Env = append(container.Env, corev1.EnvVar{
-		Name:  "DD_EKS_FARGATE",
-		Value: "true",
-	})
+	container.Resources = resourceLimits
```

Comment:
If the container has mem requests/limits defined, but the override only contains CPU requests/limits, the mem ones will be lost. Not sure if this is a problem or not, but pointing it out just in case.

Reply:
Indeed, I thought about this. 

The reason I decided to override both requests and limits is that resource requirements are considered as one config combining both requests and limits. 

If the user specifies only requests, I would assume they would not expect memory limits to be set by default. If they want to set it, it will be easy to do. If they don't want to set it, it will look weird to ask the user to set the limit for a nil value.